Marlin/src/module/stepper.cpp

```diff
@@ -2019,7 +2019,7 @@ uint32_t Stepper::block_phase_isr() {
         cutter.apply_power(current_block->cutter_power);
       #endif
 
-      TERN_(POWER_LOSS_RECOVERY, recovery.info.sdpos = current_block->sdpos);
+      TERN_(POWER_LOSS_RECOVERY, recovery.info.sdpos = Planner::block_buffer[BLOCK_MOD(Planner::block_buffer_tail - 1)].sdpos);
```

Comment:
At this point the previous block is finished and we are about to start a new `current_block`, so we *do* want the SD position from the `current_block`. It *should* contain the position of the corresponding command in the file. If we're not getting the right value here into `recovery.info.sdpos` then we should look at the code which sets the `sdpos` in the block structure instead.